evm_arithmetization/src/generation/state.rs

```diff
@@ -208,21 +211,13 @@ pub(crate) trait State<F: Field> {
                 if let Some(halt_context) = opt_halt_context {
                     if self.get_context() == halt_context {
                         // Only happens during jumpdest analysis, we don't care about the output.
-                        return Ok((final_registers, Some(final_mem)));
+                        return Ok((final_registers, final_mem));
```

Comment:
```suggestion
                        return Ok((final_registers, None));
```
Unrelated nit, but this makes a little bit more sense given the comment.